dynamiq/nodes/agents/base.py

```diff
@@ -150,6 +168,27 @@ def execute(
         logger.debug(f"Agent {self.name} - {self.id}: finished with result {result}")
         return execution_result
 
+    def _retrieve_memory(self, input_data):
+        """
+        Retrieves memory based on the selected strategy: 'relevant', 'all', or 'both'.
+        """
+        user_id = input_data.get("user_id", None)
+        filters = {"user_id": user_id} if user_id else None
+
+        if self.memory_retrieval_strategy == "relevant":
+            relevant_memory = self.memory.get_search_results_as_string(query=input_data.get("input"), filters=filters)
```

Comment:
Minor: It might be worth extracting the query to a higher level and reusing it here and later for easier code readability
`query = input_data.get("input")`